### ISSUE TITLE
Learning rate not adjusting with adaptive optimizer

### ISSUE THREAD

**Issue by VoltLoop** [kohya-ss/sd-scripts#2084]
I'm not sure what's going on, but I've been trying to train a lora using an adapative optimizer and it just doesn't seem to want to adapt the learning rates. DAdaptation and DAdaptAdam loss reaches NaN and exports black 1kb samples (dropping the LR to less than 1 fixes this). Also, the LR values on tensorboard don't seem to be adjusting as it trains, it just stays at the starting value. Not sure if tensorboard just doesn't graphy it properly, but if 1.0 is the recommended LR value, why is this happening? Prodigy does not seem to be adjusting its LR either, and I'm still using 1 for LR. No errors, just doesn't seem to work.

![Image](https://github.com/user-attachments/assets/f7b13f11-949c-4ea1-a088-f78b6519a577)

Here's my config:

> bucket_reso_steps = 64
>                          cache_latents = true
>                          caption_extension = ".txt"
>                          clip_skip = 1
>                          dynamo_backend = "no"
>                          enable_bucket = true
>                          epoch = 40
>                          flip_aug = true
>                          gradient_accumulation_steps = 1
>                          huber_c = 0.1
>                          huber_scale = 1
>                          huber_schedule = "snr"
>                          keep_tokens = 1
>                          logging_dir = "C:\\AI\\Training\\Test\\Training Set\\log"
>                          loss_type = "l2"
>                          lr_scheduler = "constant"
>                          lr_scheduler_args = []
>                          lr_scheduler_num_cycles = 1
>                          lr_scheduler_power = 1
>                          max_bucket_reso = 1024
>                          max_data_loader_n_workers = 0
>                          max_grad_norm = 1
>                          max_timestep = 1000
>                          max_token_length = 225
>                          max_train_steps = 8640
>                          min_bucket_reso = 320
>                          min_snr_gamma = 5
>                          mixed_precision = "bf16"
>                          network_alpha = 1
>                          network_args = []
>                          network_dim = 64
>                          network_module = "networks.lora"
>                          noise_offset_type = "Original"
>                          optimizer_args = []
>                          optimizer_type = "DAdaptAdam"
>                          output_dir = "C:\\AI\\Training\\Test\\Training Set\\model"
>                          output_name = "test_v1"
>                          pretrained_model_name_or_path = "C:/AI/StabilityMatrix/Data/Models/StableDiffusion/SD_realisticVisionV60B1_v51VAE.safetensors"
>                          prior_loss_weight = 1
>                          resolution = "512,512"
>                          sample_every_n_epochs = 1
>                          sample_prompts = "C:\\AI\\Training\\Test\\Training Set\\model\\sample/prompt.txt"
>                          sample_sampler = "euler_a"
>                          save_every_n_epochs = 2
>                          save_model_as = "safetensors"
>                          save_precision = "bf16"
>                          save_state_on_train_end = true
>                          seed = 1997
>                          shuffle_caption = true
>                          text_encoder_lr = [ 1.0, 1.0,]
>                          train_batch_size = 1
>                          train_data_dir = "C:\\AI\\Training\\Test\\Training Set\\img"
>                          unet_lr = 1
>                          wandb_run_name = "test_v1"
>                          xformers = true

Any help would be greatly appreciated.

**Comment by rockerBOO** [comment]
It is adapting, but once it does it causes instability and the loss goes to NaN. Could try lowering the d_coef or lower the LR and see if that works better. Also maybe try Prodigy which is a newer DAdapt like optimizer

**Comment by VoltLoop** [comment]
Thanks for the quick reply. 

Just to be clear, tensorboard doesn't actually show the adapted LR values right? If it doesn't, then that's been what's throwing me off.

**Comment by rockerBOO** [comment]
The d*lr is the adapted LR (if you use 1.0 and d is 0.008 then thats is what it is showing there)

**Comment by rockerBOO** [comment]
The D part of DAdaptation. The D*LR going to 0.008 might be a little high but you can lower the LR or d_coef for it to work. Since the optimizer is producing that d value it's something with how it is learning in general but hard to tell what it might be from your settings.
